modules/library/main/src/main/java/org/geotools/data/crs/ReprojectFeatureReader.java

```diff
@@ -162,9 +190,19 @@ public SimpleFeature next()
         Object[] attributes = next.getAttributes().toArray();
 
         try {
-            for (int i = 0; i < attributes.length; i++) {
-                if (attributes[i] instanceof Geometry) {
-                    attributes[i] = transformer.transform((Geometry) attributes[i]);
+            for (int i = 0; i < schema.getDescriptors().size(); i++) {
+                if (schema.getDescriptor(i) instanceof GeometryDescriptor
+                        && attributes[i] != null) {
```

Comment:
Since ``attributes[i]`` is cast to Geometry a few lines down, the check for the value being a Geometry, rather than not null, is safer (I do realize a geometry description should have a Geometry value).

Reply:
Ok.